File: shopapp-angular/src/app/dtos/order.dto.ts
import { CartItemDTO } from './cart.item.dto';

export class OrderDTO {
  user_id: number;
  phone_number: string;
  address: string;
  note: string;
  fullname: string;
  email: string;
  payment_method: string;
  shipping_method: string;
  total_money: number;
  coupon_code: string;
  cart_items: CartItemDTO[];
  constructor(data: any) {
    this.user_id = data.user_id;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.fullname = data.fullname;
    this.email = data.email;
    this.payment_method = data.payment_method;
    this.shipping_method = data.shipping_method;
    this.coupon_code = data.coupon_code;
    this.total_money = data.total_money;
    this.cart_items = data.cart_items;
  }
}

File: src/main/java/com/ngleanhvu/shopapp/dto/OrderDTO.java
package com.ngleanhvu.shopapp.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.util.List;

@Builder
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderDTO {
    Integer id;
    @Min(value = 1, message = "User's id must be > 0")
    @JsonProperty("user_id")
    Integer userId;
    @JsonProperty("fullname")
    String fullName;
    String email;
    String address;
    @JsonProperty("phone_number")
    @NotBlank(message = "Phone number is required")
    String phoneNumber;
    String note;
    @Min(value = 0, message = "Total money must be >= 0")
    @JsonProperty("total_money")
    float totalMoney;
    @JsonProperty("shipping_address")
    String shippingAddress;
    @JsonProperty("shipping_method")
    String shippingMethod;
    @JsonProperty("payment_method")
    String paymentMethod;
    @JsonProperty("shipping_date")
    LocalDate shippingDate;
    @JsonProperty("cart_items")
    List<CartItemDTO> cartItemDTOs;
    @JsonProperty("status")
    String status;
}
